prefix-map.c

```diff
@@ -0,0 +1,111 @@
+#include "cache.h"
+#include "prefix-map.h"
+
+static int map_cmp(const void *unused_cmp_data,
+		   const void *entry,
+		   const void *entry_or_key,
+		   const void *unused_keydata)
+{
+	const struct prefix_map_entry *a = entry;
+	const struct prefix_map_entry *b = entry_or_key;
+
+	return a->prefix_length != b->prefix_length ||
+		strncmp(a->name, b->name, a->prefix_length);
```

Comment:
Here is also indentation misplacement.

Reply:
This is actually intentional. I know, Git's indentation conventions are confusing (and inconsistent): for function calls, or for conditional expressions, we want to continue one column after the opening parenthesis. But this is not such a case, so we indent with an extra tab.

Reply:
Okay, thanks for still teaching me about conventions :)